File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> sum = new HashMap<>();
        int[] indices = new int[2];
        for(int i = 0;i < nums.length;i++){
            sum.put(nums[i],i);
        }
        int remaining = 0;
        for(int i = 0;i < nums.length;i++){
            remaining = target - nums[i];
            sum.remove(nums[i],i);
            if(sum.containsKey(remaining)){
                indices[0] = i;
                indices[1] = sum.get(remaining);
                break;
            }
        }
        return indices;
    }
}

File: Data Structures & Algorithms/binary-search/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        left = 0
        right = len(nums)-1
        n = (left + right+1)//2
        while left <= right:
            if(nums[n] == target):
                return n
            elif target > nums[n]:
                left = n +1
                n = (left + right)//2
        
            else:
                right = n-1
                n = (left + right)//2
        return -1


        